core/src/engine/coro/pool.hpp

```diff
@@ -111,39 +126,74 @@ Pool<Task>::~Pool() = default;
 
 template <typename Task>
 typename Pool<Task>::CoroutinePtr Pool<Task>::GetCoroutine() {
-  struct CoroutineMover {
-    std::optional<Coroutine>& result;
-
-    CoroutineMover& operator=(Coroutine&& coro) {
-      result.emplace(std::move(coro));
-      return *this;
-    }
-  };
-
   std::optional<Coroutine> coroutine;
-  CoroutineMover mover{coroutine};
 
   // First try to dequeue from 'working set': if we can get a coroutine
   // from there we are happy, because we saved on minor-page-faulting (thus
   // increasing resident memory usage) a not-yet-de-virtualized coroutine stack.
-  if (used_coroutines_.try_dequeue(
-          GetUsedPoolToken<moodycamel::ConsumerToken>(), mover) ||
-      initial_coroutines_.try_dequeue(mover)) {
-    --idle_coroutines_num_;
+  if (local_coro_buffer<Task>.size() > 0) {
+    coroutine = std::move(local_coro_buffer<Task>.back());
+    local_coro_buffer<Task>.pop_back();
   } else {
-    coroutine.emplace(CreateCoroutine());
+    const std::size_t deque_num =
+        std::min(idle_coroutines_num_.load(std::memory_order_acquire),
+                 kLocalCoroutineMoveSize);
+    if (deque_num > 0) {
```

Comment:
Let's extract this into a method: `bool TryPopulateLocalCache()`
It will also modify `idle_coroutines_num_`

This way we can use a nice if-else chain:

```cpp
if (local_coro_buffer<Task>.size() > 0) {
  ...
} else if (TryPopulateLocalCache()) {
  ...
} else {
  ...
}
```